### PR TITLE
Shared tria artificial bugfix

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/dealii/dealii/issues/1818

### CODE REVIEW

File: tests/sharedtria/dof_01.cc [hunk 1]
```diff
@@ -92,19 +92,20 @@ void test()
...
-          deallog << v[i] << " ";
-          sum += v[i];
-        }
-      deallog << " sum: " << sum << std::endl;
+          << "n_dofs: " << dof_handler.n_dofs() << std::endl;
+//          << "n_locally_owned_dofs: " << dof_handler.n_locally_owned_dofs() << std::endl;
+
+//      deallog << "n_locally_owned_dofs_per_processor: ";
+//      std::vector<types::global_dof_index> v = dof_handler.n_locally_owned_dofs_per_processor();
+//      unsigned int sum = 0;
```
Comment (bangerth) [inline tests/sharedtria/dof_01.cc]: I know why you take all of these lines out, but I think the test no longer makes a lot of sense without this output -- there is simply nothing left that's indicative whether the class works or not. So my suggestion is that we leave this in here for now and find ways to deal with it.

**Comment by bangerth** [comment]
The first two commits here are uncontroversial and we should merge them. I do think that we should leave the tests as is, as mentioned in my comment above.

File: tests/sharedtria/dof_01.cc (hunk 1 above)
Comment (davydden) [inline tests/sharedtria/dof_01.cc]: actually there are a number of asserts below which test for intersection if indexsets and that the sum of all dofs is the total dofs. So i would say we have some fair amount of checks.
Triangulation tests have less things to check, but I also added a couple of Asserts there.

I if you still prefer to keep it as it is, i am fine with that as well. Let me know.

Comment (bangerth) [inline tests/sharedtria/dof_01.cc]: Ah, I didn't see this -- thanks for pointing it out. In that case I'm ok with it.

Do you think it's worthwhile keeping this text commented out? Would you ever want to bring them back in? If not, take the commented lines out and merge yourself. If you think we may eventually want to bring them back in (e.g., if we replace the partitioning code by something homegrown) then just merge as is.

Comment (davydden) [inline tests/sharedtria/dof_01.cc]: i would leave it just in case a) there is a homegrown partition via p4est; b) parmetis is more predictable.

**Comment by davydden** [comment]
i will squash the last 2 commits which deal with unit tests in one and keep the first one separately as it deals with the bug.

**Pull request merged**